server/lease/lessor_test.go

```diff
@@ -207,6 +207,29 @@ func TestLessorRevoke(t *testing.T) {
 	}
 }
 
+func renew(t *testing.T, le *lessor, id LeaseID) int64 {
+	ch := make(chan int64, 1)
+	errch := make(chan error, 1)
+	go func() {
+		ttl, err := le.Renew(id)
+		if err != nil {
+			errch <- err
+		} else {
+			ch <- ttl
+		}
+	}()
+
+	select {
+	case ttl := <-ch:
+		return ttl
+	case err := <-errch:
+		t.Fatalf("failed to renew lease (%v)", err)
+	case <-time.After(2 * time.Second):
+		t.Fatal("timed out while renewing lease")
+	}
+	panic("unreachable")
+}
```

Comment:
I am thinking [testify](https://github.com/stretchr/testify) should support this, but unfortunately I do not see such function.  The similar one is [Eventually](https://github.com/stretchr/testify/blob/1333b5d3bda8cf5aedcf3e1aaa95cac28aaab892/require/require.go#L357), but it can't satisfy this case. Probably you can deliver a PR for [testify](https://github.com/stretchr/testify) to support it, and enhance the test case in future.